doc/vlan/SONiC_vlan_translation_qnq.md

```diff
@@ -0,0 +1,1025 @@
+              |     +--rw oc-intf-ext:config
+              |     |  +--rw oc-intf-ext:inner-vlan-id?   oc-vlan-types:vlan-id
+              |     |  +--rw oc-intf-ext:outer-vlan-id?   oc-vlan-types:vlan-id
+              |     +--ro oc-intf-ext:state
+              |        +--ro oc-intf-ext:inner-vlan-id?   oc-vlan-types:vlan-id
+              |        +--ro oc-intf-ext:outer-vlan-id?   oc-vlan-types:vlan-id
+              +--rw oc-intf-ext:ingress-mapping
+              |  +--rw oc-intf-ext:config
+              |  |  +--rw oc-intf-ext:vlan-stack-action?   matched-vlan-stack-action
+              |  +--ro oc-intf-ext:state
+              |     +--ro oc-intf-ext:vlan-stack-action?   matched-vlan-stack-action
+              +--rw oc-intf-ext:egress-mapping
+                 +--rw oc-intf-ext:config
+                 |  +--rw oc-intf-ext:vlan-stack-action?   matched-vlan-stack-action
+                 +--ro oc-intf-ext:state
+                    +--ro oc-intf-ext:vlan-stack-action?   matched-vlan-stack-action
+```
+
+#### 3.6.1.2 SONiC Yang
+```diff
+module: sonic-vlan-stacking
+  +--rw sonic-vlan-stacking
+     +--rw VLAN_STACKING
+        +--rw VLAN_STACKING_LIST* [ifname s_vlan stage]
+           +--rw ifname             union
+           +--rw s_vlan             string
+           +--rw stage              scommon:stack_stage
+           +--rw c_vlanids*         union
+           +--rw s_vlan_priority?   uint8
+
+
+module: sonic-vlan-translation
+  +--rw sonic-vlan-translation
+     +--rw VLAN_TRANSLATION
+        +--rw VLAN_TRANSLATION_LIST* [interface_name s_vlan stage]
+           +--rw interface_name    union
+           +--rw s_vlan            string
+           +--rw stage             scommon:stack_stage
+           +--rw c_vlanid_inner?   uint16
+           +--rw c_vlanid_outer?   uint16
+
+```
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+```
+Syntax:
+[no] switchport vlan-mapping  {{<cvlan-id> [inner <inner-cvlan-id>]}  | 
+                               {{<cvlan_list> | { add | remove} <cvlan-list>} dot1q-tunnel } <svlan-id>
+```
+
+#### 3.6.2.1.1 VLAN Translation Commands
+
+#### Single Tag Translation
+```
+[no] switchport vlan-mapping <cvlan-id> <svlan-id>
+```
+Example
+```
+sonic# configure terminal 
+sonic(config)# interface Ethernet 4
+sonic(conf-if-Ethernet4)# switchport 
+  access        Set access mode characteristics of the interface
+  trunk         Configure trunking parameters on an interface
+  vlan-mapping  Configure VLAN translation parameters on an interface
+
+sonic(conf-if-Ethernet4)# switchport vlan-mapping 
+  <1..4094>   Configure VLAN mapping customer VLAN ID
+  add         Configure VLAN stacking match customer VLANs on an interface
+  remove      Remove VLAN stacking match customer VLANs on an interface
+  VLAN-LIST   (-) or (,) separated individual VLAN IDs and ranges of VLAN IDs; for example, 20,70-100,142
+
+sonic(conf-if-Ethernet4)# switchport vlan-mapping 100 
```

Comment:
dot1q-tunnel keyword  will applicable for single vlan or list of vlans or both? incase of single tag translation , can the user input VLAN-LIST or is it mandatory to have only single vlan?

Reply:
dot1Q-tunnel/stacking : Can tunnel a single or list or range of vlans under an SVLAN (Ingress PUSH, Egress POP)
VLANS Translation :   single cvlan or a stack (double tag) can be translated to an SVLAN (Ingress & Egress SWAP)